forth.asm

```diff
@@ -203,6 +203,28 @@ dict_entry tonumber, '>NUMBER', 0x0 ;( str -- n)
     push    ebx
     jmp     next
 
+interpret_if:
+    ; if stack nozero, advance instruction pointer
+    pop     eax
+    test    eax, eax
+    jnz     .true
+    add     esi, 0x8
+    jmp     next
+.true:
+    jmp     next
+
+
+dict_entry interpret, 'INTERPRET-WORD', 0x0 ; ( str -- )
+    call    docolon
+    dd      blank
+    dd      cword
+    dd      find
+    dd      interpret_if
```

Comment:
Thanks for the suggestion. I think that would be confusing though, because the labels are not the actual forth words. For example `blank` will be `BL` in forth, or `star` is `*`.

Maybe I phrased it confusing, because I'm not even sure what I am looking for.
Basically I would like to have something like this, i.e. inline the `interpret_if`:
```
    dd      blank
    dd      cword
    dd      find
    pop     eax
    test    eax, eax
    jnz     .execute
    dd     tonumber
    dd     exit
.execute:
    dd     execute
    dd     exit
```
which is of course not possible. I might have to look at other implementations how they did it. I think there must be something like an "internal `next`".